src/plugins/genericScheme/proposal.ts

```diff
@@ -0,0 +1,168 @@
+            genericScheme {
+              id
+              contractToCall
+              callData
+              executed
+              returnValue
+            }
+          }
+        `
+      }
+    }
+    return this.fragmentField
+  }
+
+  public static itemMap(
+    context: Arc,
+    item: any,
+    query: DocumentNode
+  ): IGenericSchemeProposalState | null {
+    if (!item) {
+      Logger.debug(`GenericScheme Proposal ItemMap failed. Query: ${query.loc?.source.body}`)
+      return null
+    }
+
+    const genericSchemeState = GenericScheme.itemMap(context, item.scheme, query)
+
+    if (!genericSchemeState) {
+      return null
+    }
+
+    const genericScheme = new GenericScheme(context, genericSchemeState)
+    const genericSchemeProposal = new GenericSchemeProposal(context, item.id)
+
+    const baseState = Proposal.itemMapToBaseState(
+      context,
+      item,
+      genericScheme,
+      genericSchemeProposal,
+      'GenericScheme'
+    )
+
+    if (!baseState) {
+      return null
+    }
+
+    return {
+      ...baseState,
+      callData: item.genericScheme.callData,
+      contractToCall: item.genericScheme.contractToCall,
+      executed: item.genericScheme.executed,
+      returnValue: item.genericScheme.returnValue
+    }
+  }
+
+  private static fragmentField: { name: string; fragment: DocumentNode } | undefined
+
+  public state(apolloQueryOptions: IApolloQueryOptions): Observable<IGenericSchemeProposalState> {
+    const query = gql`query ProposalState
+      {
+        proposal(id: "${this.id}") {
+          ...ProposalFields
+          votes {
+            id
+          }
+          stakes {
+            id
+          }
+        }
+      }
+      ${Proposal.baseFragment}
+      ${Plugin.baseFragment}
+    `
+
+    const result = this.context.getObservableObject(
+      this.context,
+      query,
+      GenericSchemeProposal.itemMap,
+      apolloQueryOptions
+    ) as Observable<IGenericSchemeProposalState>
+    return result
+  }
+
+  public redeemerContract() {
+    for (const version of REDEEMER_CONTRACT_VERSIONS) {
+      try {
+        const contractInfo = this.context.getContractInfoByName('Redeemer', version)
+        return this.context.getContract(contractInfo.address)
+      } catch (err) {
+        if (!err.message.match(/no contract/i)) {
+          // if the contract cannot be found, try the next one
+          throw err
+        }
+      }
+    }
+    throw Error(
+      `No Redeemer contract could be found (search for versions ${REDEEMER_CONTRACT_VERSIONS})`
+    )
+  }
+
+  public redeemRewards(beneficiary?: Address): Operation<boolean> {
+    const mapReceipt = (receipt: ITransactionReceipt) => true
+
+    const createTransaction = async (): Promise<ITransaction> => {
+      if (!beneficiary) {
+        beneficiary = NULL_ADDRESS
+      }
+
+      const state = await this.fetchState()
+      const pluginAddress = this.context.getContractInfoByName(
+        'ContributionReward',
```

Comment:
This cannot be right - "contributionreward" in the middle of the genericScheme code :-)